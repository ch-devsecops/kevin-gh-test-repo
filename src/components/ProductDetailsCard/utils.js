import { ENGINE_SITE_NAME, MARINE_SITE_NAME } from '../../utils/constants';
import { getGtmTagValue } from '../../utils/gtmEvents';

/**
 * Creates GTM attributes for a productName
 * @param {string} productName
 * @param {object} gtmTags
 * @returns gtmAttributes
 */
export const getProductLineGtmTags = (productName, gtmTags) => {
  switch (productName) {
    case MARINE_SITE_NAME:
      return Object.freeze({
        'data-gtm-interaction-type': 'cta: find a dealer',
        'data-gtm-component-type': getGtmTagValue(gtmTags?.componentName),
        'data-gtm-title': 'find a dealer',
      });
    default:
      return Object.freeze({
        'data-gtm-interaction-type': getGtmTagValue(gtmTags?.interactionType),
        'data-gtm-title': getGtmTagValue(gtmTags?.title),
        'data-gtm-component-type': getGtmTagValue(gtmTags?.componentName),
        'data-gtm-body-style': getGtmTagValue(gtmTags?.gtmSeriesName),
        'data-gtm-model': getGtmTagValue(gtmTags?.gtmCrankshaftName),
        'data-gtm-trim': getGtmTagValue(gtmTags?.gtmTrimName),
      });
  }
};

/**
 * Creates product line gtm tags
 * @productName {string} The name of the productLine used
 * @gtmTags {object} The gtmTags object
 * @checkCompare {boolean} The status of the compare button
 * @returns The gtmTags for productLine.
 */
export const getAddToCompareProductLineGtmTags = (appName, gtmTags, checkCompare) => {
  // eslint-disable-next-line default-case
  switch (appName) {
    case MARINE_SITE_NAME:
      return Object.freeze({
        'data-gtm-component-type': getGtmTagValue(gtmTags?.componentName),
        'data-gtm-interaction-type': 'cta: compare',
        'data-gtm-title': getGtmTagValue(gtmTags?.addTitle),
      });
    case ENGINE_SITE_NAME:
      return Object.freeze({
        'data-gtm-interaction-type': getGtmTagValue(gtmTags?.removeInteractionType),
        'data-gtm-title': getGtmTagValue(checkCompare ? gtmTags?.removeTitle : gtmTags?.addTitle),
        'data-gtm-component-type': getGtmTagValue(gtmTags?.componentName),
        'data-gtm-body-style': getGtmTagValue(gtmTags?.gtmSeriesName),
        'data-gtm-model': getGtmTagValue(gtmTags?.gtmCrankshaftName),
        'data-gtm-trim': getGtmTagValue(gtmTags?.gtmTrimName),
      });
  }
};
