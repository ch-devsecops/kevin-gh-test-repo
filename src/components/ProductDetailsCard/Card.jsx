import React, { useContext, useEffect, useState } from 'react';
import {
  Box,
  Button,
  Column,
  Copy,
  H1,
  Icon,
  Image,
  MarkdownHeading,
  Optional,
  Row,
  UnorderedList,
  UnorderedListItem,
  HangTag,
} from '@honda-canada/design-system-react';
import formatPrice from '@honda-canada/js-utilities/lib/formatPrice';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import EmbeddedCarousel from './EmbeddedCarousel';
import themeStyles from './ProductDetailsCard.styles';
import LocalStorageContext from '../LocalStorageContext/LocalStorageContext';
import { compareProducts } from '../../utils/constants';
import { styledCompiler } from '../../utils/markdown';
import { getAddToCompareProductLineGtmTags, getProductLineGtmTags } from './utils';
import useAppName from '../../utils/sitecoreContext/useAppName';
import { useLanguage } from '../../utils/sitecoreContext';

const PrimaryButton = themeStyles.apply(Button, 'PrimaryButton');
const SecondaryToggleButton = themeStyles.apply(Button, 'SecondaryToggleButton');
const List = themeStyles.apply(UnorderedList, 'List');
const ListItem = themeStyles.apply(UnorderedListItem, 'ListItem');
const ButtonsGroup = themeStyles.apply(Box, 'ButtonsGroup');
const Title = themeStyles.apply(H1, 'Title');
const Subtitle = themeStyles.apply(MarkdownHeading, 'Subtitle');
const BodyText = themeStyles.apply(Copy, 'BodyText');
const Heading = themeStyles.apply(MarkdownHeading, 'Heading');
const LeftColumn = themeStyles.apply(Column, 'LeftColumn');
const RightColumn = themeStyles.apply(Column, 'RightColumn');
const RightColumnWrapper = themeStyles.apply(Box, 'RightColumnWrapper');
const ImageContainer = themeStyles.apply(Box, 'ImageContainer');
const HangTagWrapper = themeStyles.apply(Box, 'HangTagWrapper');

const ProductDetailsCard = ({
  detId,
  title,
  subtitle,
  tagline,
  list,
  images,
  gtmTags,
  headingTitle,
  actionButtonText,
  compareIconName,
  PriceComponent,
  prices,
}) => {
  const { t } = useTranslation();
  const language = useLanguage();
  const { toCompareProducts, setToCompareProducts, removeFromCompareProducts } = useContext(LocalStorageContext);
  const [checkCompare, setCheckCompare] = useState(false);
  const [isCompareDisabled, setIsCompareDisabled] = useState(false);

  const appName = useAppName();
  const iconName = checkCompare ? 'close' : compareIconName;

  const discountValue = prices?.discount?.priceDiscountAmount;

  useEffect(() => {
    setIsCompareDisabled(toCompareProducts.length >= compareProducts.limit && !toCompareProducts.includes(detId));
    setCheckCompare(toCompareProducts.includes(detId));
  }, [detId, toCompareProducts]);

  const imagesList = images?.map((image, i) => (
    <ImageContainer data-testid="product-image-container" key={i.toString()}>
      <Image
        width="auto"
        data-testid="product-image"
        src={image.url}
        alt={image.alt?.value}
        coverContainer
        loading="eager"
        fetchpriority="high"
      />
    </ImageContainer>
  ));

  const onCompareEngine = () => {
    setCheckCompare(!checkCompare);
    if (checkCompare) {
      removeFromCompareProducts(detId);
    } else {
      setToCompareProducts(detId);
    }
  };

  return (
    <Row>
      <LeftColumn width={[1, 1, 1 / 2]}>
        <Optional when={!!discountValue}>
          <HangTagWrapper>
            <HangTag
              label={t('Shared.Common.saveLabel')}
              amount={formatPrice(discountValue, language)}
              language={language}
            />
          </HangTagWrapper>
        </Optional>
        <Optional when={imagesList}>
          <EmbeddedCarousel mt={['s', 'l']} items={imagesList} />
        </Optional>
      </LeftColumn>
      <RightColumn width={[1, 1, 1 / 2]}>
        <RightColumnWrapper>
          <Title data-testid="product-title">{title}</Title>
          <Subtitle headingOverride="h5" data-testid="product-subtitle">
            {subtitle}
          </Subtitle>
          <BodyText data-testid="tagline">{styledCompiler(tagline)}</BodyText>
          <Optional when={Boolean(list?.length)}>
            <Heading headingOverride="h6" data-testid="heading-title">
              {headingTitle}
            </Heading>
          </Optional>
          <List bulletStyle="bullet" data-testid="product-features">
            {list?.map((item, i) => (
              <ListItem data-testid="product-feature" key={i.toString()}>
                {item}
              </ListItem>
            ))}
          </List>
          <Optional when={Boolean(PriceComponent)}>
            <PriceComponent />
          </Optional>
          <ButtonsGroup>
            <PrimaryButton
              as="a"
              data-testid="primary-button"
              styling="primary"
              href={t('Shared.Common.dealersPageUrl')}
              {...getProductLineGtmTags(appName, gtmTags)}
            >
              {actionButtonText}
            </PrimaryButton>
            <SecondaryToggleButton
              as="button"
              styling="secondary"
              onClick={onCompareEngine}
              data-testid="secondary-button"
              disabled={isCompareDisabled}
              {...getAddToCompareProductLineGtmTags(appName, gtmTags, checkCompare)}
            >
              <Icon name={iconName} iconColor={isCompareDisabled ? 'grey.2' : 'secondary'} className={iconName} />
              {t(checkCompare ? 'Shared.Common.removeFromCompareButton' : 'Shared.Common.addToCompareButton')}
            </SecondaryToggleButton>
          </ButtonsGroup>
        </RightColumnWrapper>
      </RightColumn>
    </Row>
  );
};

ProductDetailsCard.propTypes = {
  actionButtonText: PropTypes.string,
  compareIconName: PropTypes.string,
  PriceComponent: PropTypes.func,
  headingTitle: PropTypes.string,
  detId: PropTypes.string,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  tagline: PropTypes.string,
  list: PropTypes.arrayOf(PropTypes.string),
  images: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      url: PropTypes.string,
      alt: PropTypes.shape({
        value: PropTypes.string,
      }),
      title: PropTypes.shape({
        value: PropTypes.string,
      }),
    }),
  ),
  gtmTags: PropTypes.shape({
    title: PropTypes.string,
    addTitle: PropTypes.string,
    removeTitle: PropTypes.string,
    gtmTrimName: PropTypes.string,
    gtmSeriesName: PropTypes.string,
    gtmCrankshaftName: PropTypes.string,
    componentName: PropTypes.string,
    interactionType: PropTypes.string,
    removeInteractionType: PropTypes.string,
  }),
};

export default ProductDetailsCard;
