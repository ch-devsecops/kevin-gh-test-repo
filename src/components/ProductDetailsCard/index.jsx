import React from 'react';
import PropTypes from 'prop-types';
import { Wrapper, Box, Column, Media } from '@honda-canada/design-system-react';
import Card from './Card';
import themeStyles from './ProductDetailsCard.styles';

const Container = themeStyles.apply(Box, 'Container');
const MobileColumn = themeStyles.apply(Column, 'MobileColumn');

const ProductDetailsCardWrapper = ({
  detId,
  title,
  subtitle,
  tagline,
  list,
  images,
  margins,
  gtmTags,
  actionButtonText,
  headingTitle,
  compareIconName,
  PriceComponent,
  prices,
}) => {
  const cardProps = {
    detId,
    title,
    subtitle,
    tagline,
    list,
    images,
    actionButtonText,
    headingTitle,
    compareIconName,
    PriceComponent,
    prices,
  };

  return (
    <Wrapper data-gtm-component-type={gtmTags?.componentName}>
      <Container data-testid="container" margins={margins}>
        <Media greaterThanOrEqual="desktop">
          <Card gtmTags={gtmTags} {...cardProps} />
        </Media>
        <Media lessThan="desktop">
          <MobileColumn>
            <Card gtmTags={gtmTags} {...cardProps} />
          </MobileColumn>
        </Media>
      </Container>
    </Wrapper>
  );
};

ProductDetailsCardWrapper.propTypes = {
  compareIconName: PropTypes.string,
  PriceComponent: PropTypes.func,
  actionButtonText: PropTypes.string,
  headingTitle: PropTypes.string,
  detId: PropTypes.string,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  tagline: PropTypes.string,
  list: PropTypes.arrayOf(PropTypes.string),
  images: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      url: PropTypes.string,
      alt: PropTypes.shape({
        value: PropTypes.string,
      }),
      title: PropTypes.shape({
        value: PropTypes.string,
      }),
    }),
  ),
  margins: PropTypes.shape({
    horizontalMargin: PropTypes.string,
    topMargin: PropTypes.string,
    bottomMargin: PropTypes.string,
    horizontalMarginMob: PropTypes.string,
    topMarginMob: PropTypes.string,
    bottomMarginMob: PropTypes.string,
  }),
  gtmTags: PropTypes.shape({
    title: PropTypes.string,
    addTitle: PropTypes.string,
    removeTitle: PropTypes.string,
    gtmTrimName: PropTypes.string,
    gtmSeriesName: PropTypes.string,
    gtmCrankshaftName: PropTypes.string,
    componentName: PropTypes.string,
    interactionType: PropTypes.string,
    removeInteractionType: PropTypes.string,
  }),
};

export default ProductDetailsCardWrapper;
