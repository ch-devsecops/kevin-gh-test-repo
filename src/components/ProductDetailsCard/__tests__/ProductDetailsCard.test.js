import React from 'react';
import { axe, cleanup, fireEvent, jssRender } from 'test-utils';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import ProductDetailsCard from '..';
import { ENGINE_SITE_NAME } from '../../../utils/constants';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/product-details-card/en.json');

const sitecoreContextMock = { site: { name: ENGINE_SITE_NAME } };

describe('<ProductDetailsCard>', () => {
  it('should render correctly', () => {
    const { container } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductDetailsCard {...mockData} />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  it('should have no accessibility violations', async () => {
    const { container } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductDetailsCard {...mockData} />
      </SitecoreContext>,
    );

    expect(await axe(container)).toHaveNoViolations();
  });

  it('should render buttons with correct attributes and correct functionality', async () => {
    const { getAllByTestId, getAllByRole } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductDetailsCard {...mockData} />
      </SitecoreContext>,
    );

    const findDistributor = getAllByRole('link')[0];
    let compareEngine = getAllByTestId('secondary-button')[0];

    expect(findDistributor).not.toHaveAttribute('disabled');
    expect(findDistributor).toHaveAttribute('href', 'Shared.Common.dealersPageUrl');

    expect(compareEngine).not.toHaveAttribute('disabled');
    expect(compareEngine).toHaveTextContent('Shared.Common.addToCompareButton');
    await fireEvent.click(compareEngine);
    compareEngine = getAllByTestId('secondary-button')[0];
    expect(compareEngine).toHaveTextContent('Shared.Common.removeFromCompareButton');
  });

  it('should render with correct margins coming from sitecore', () => {
    const { queryByTestId } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductDetailsCard {...mockData} />
      </SitecoreContext>,
    );
    const container = queryByTestId('container');
    expect(container).toHaveStyle('margin-top:20px');
    expect(container).toHaveStyle('margin-bottom:30px');
    expect(container).toHaveStyle('margin-left:40px');
    expect(container).toHaveStyle('margin-right:40px');
  });

  it('should render with default margins when no margins coming from sitecore', () => {
    const { queryByTestId } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductDetailsCard {...{ ...mockData, margins: {} }} />
      </SitecoreContext>,
    );
    const container = queryByTestId('container');
    expect(container).toHaveStyle('margin-top:l');
    expect(container).toHaveStyle('margin-bottom:l');
    expect(container).toHaveStyle('margin-left:10px');
    expect(container).toHaveStyle('margin-right:10px');
  });

  it('should have gtm tags with correct data', async () => {
    const { container, getAllByTestId, getAllByRole } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductDetailsCard {...mockData} />
      </SitecoreContext>,
    );

    const findDistributor = getAllByRole('link')[0];
    const compareEngine = getAllByTestId('secondary-button')[0];

    expect(container.firstChild).toHaveAttribute('data-gtm-component-type', 'EngineDetailsCard');
    expect(findDistributor).toHaveAttribute('data-gtm-interaction-type', 'find distributor');
    expect(findDistributor).toHaveAttribute('data-gtm-title', 'Find a distributor');
    expect(findDistributor).toHaveAttribute('data-gtm-body-style', 'HorizontalCrankshaft');
    expect(findDistributor).toHaveAttribute('data-gtm-model', 'GX Series');
    expect(findDistributor).toHaveAttribute('data-gtm-trim', 'GX100');

    expect(compareEngine).toHaveAttribute('data-gtm-interaction-type', 'comparison results');
    expect(compareEngine).toHaveAttribute('data-gtm-title', 'item added');
    expect(compareEngine).toHaveAttribute('data-gtm-body-style', 'HorizontalCrankshaft');
    expect(compareEngine).toHaveAttribute('data-gtm-model', 'GX Series');
    expect(compareEngine).toHaveAttribute('data-gtm-trim', 'GX100');
  });

  it('should not have tags when no tags data provided', async () => {
    const { container, getAllByTestId, getAllByRole } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductDetailsCard {...{ ...mockData, gtmTags: {} }} />
      </SitecoreContext>,
    );

    const findDistributor = getAllByRole('link')[0];
    const compareEngine = getAllByTestId('secondary-button')[0];

    expect(container.firstChild).not.toHaveAttribute('data-gtm-component-type', 'EngineDetailsCard');
    expect(findDistributor).not.toHaveAttribute('data-gtm-interaction-type');
    expect(findDistributor).not.toHaveAttribute('data-gtm-title');
    expect(findDistributor).not.toHaveAttribute('data-gtm-body-style');
    expect(findDistributor).not.toHaveAttribute('data-gtm-model');
    expect(findDistributor).not.toHaveAttribute('data-gtm-trim');

    expect(compareEngine).not.toHaveAttribute('data-gtm-interaction-type');
    expect(compareEngine).not.toHaveAttribute('data-gtm-title');
    expect(compareEngine).not.toHaveAttribute('data-gtm-body-style');
    expect(compareEngine).not.toHaveAttribute('data-gtm-model');
    expect(compareEngine).not.toHaveAttribute('data-gtm-trim');
  });
});
