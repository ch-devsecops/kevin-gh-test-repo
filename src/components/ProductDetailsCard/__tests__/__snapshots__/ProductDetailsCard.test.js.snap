// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailsCard> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 gXilRl"
    data-gtm-component-type="EngineDetailsCard"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jPwEai"
      data-testid="container"
    >
      <div
        class="fresnel-container fresnel-greaterThanOrEqual-desktop "
      >
        <div
          class="Box-sc-1bsf784-0 jGVkdm"
          display="flex"
          width="100%"
        >
          <div
            class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 sgibs"
            width="1,1,0.5"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 hQILzn"
            >
              <div
                class="Box-sc-1bsf784-0 isOmPB sc-1k44dfv-0 ednPAD"
                display="flex"
              >
                <div
                  class="Box-sc-1bsf784-0 gaxtDi"
                  display="flex"
                  width="100%"
                >
                  <div
                    class="Box-sc-1bsf784-0 Column-t2p57o-0 hVRYAI"
                    display="flex"
                    width="0,0.125"
                  />
                  <div
                    class="Box-sc-1bsf784-0 Column-t2p57o-0 jcWdVb"
                    display="flex"
                    width="1,0.75"
                  >
                    <div
                      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jazOpQ"
                    >
                      <div
                        aria-live="polite"
                        class="Box-sc-1bsf784-0 sc-1k44dfv-0 daGqcZ"
                        height="157px,362px,402px"
                      >
                        <div
                          class="Box-sc-1bsf784-0 sc-1k44dfv-0 jyKEsn"
                        >
                          <div
                            class="Box-sc-1bsf784-0 sc-1k44dfv-0 dlUJcO"
                            data-testid="product-image-container"
                          >
                            <img
                              alt="Front of Honda GX100 engine"
                              class="sc-1k44dfv-0 cFTYBj"
                              data-testid="product-image"
                              draggable="true"
                              fetchpriority="high"
                              loading="eager"
                              src="/sample-mock-data/media/img/product-card/engineImage.png"
                              width="auto"
                            />
                          </div>
                        </div>
                        <div
                          class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
                        >
                          <div
                            class="Box-sc-1bsf784-0 sc-1k44dfv-0 dlUJcO"
                            data-testid="product-image-container"
                          >
                            <img
                              alt="Side of Honda GX100 engine"
                              class="sc-1k44dfv-0 cFTYBj"
                              data-testid="product-image"
                              draggable="true"
                              fetchpriority="high"
                              loading="eager"
                              src="/sample-mock-data/media/img/product-details-card/engineDetailsImage.png"
                              width="auto"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Box-sc-1bsf784-0 Column-t2p57o-0 hVRYAI"
                    display="flex"
                    width="0,0.125"
                  />
                </div>
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 elVxZQ"
              >
                <div
                  aria-label="Shared.Common.previousSlideAria"
                  class="IconWrapper-lp9q2r-0 hIeNMk"
                  cursor="pointer"
                  data-testid="arrow-left"
                  display="flex"
                  role="button"
                  tabindex="0"
                />
                <div
                  class="Box-sc-1bsf784-0 sc-1k44dfv-0 evNJbO"
                >
                  <button
                    aria-label="0"
                    class="sc-1k44dfv-0 cjSExO"
                    data-testid="step1"
                    role="button"
                  />
                  <button
                    aria-label="1"
                    class="sc-1k44dfv-0 ipUiMI"
                    data-testid="step2"
                    role="button"
                  />
                </div>
                <div
                  aria-label="Shared.Common.nextSlideAria"
                  class="IconWrapper-lp9q2r-0 hIeNMk"
                  cursor="pointer"
                  data-testid="arrow-right"
                  display="flex"
                  role="button"
                  tabindex="0"
                />
              </div>
            </div>
          </div>
          <div
            class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 cVEbMC"
            width="1,1,0.5"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 krtBLR"
            >
              <h1
                class="sc-1k44dfv-0 sc-1k44dfv-0 eNZGfe"
                data-testid="product-title"
              >
                GX100
              </h1>
              <div
                class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 jVHFQZ"
                data-testid="product-subtitle"
              >
                

                <h5
                  class="sc-1k44dfv-0 gXWcpP"
                >
                  Horizontal Crankshaft
                </h5>
              </div>
              <p
                class="sc-1k44dfv-0 sc-1k44dfv-0 iZbHgV"
                color="typographyDefault"
                data-testid="tagline"
                font-family="default"
                font-weight="regular"
              >
                <span>
                  Day in, day out, get solid performance, reliability and fuel efficiency. Ideal for lawn, construction and agriculture equipment.
                </span>
              </p>
              <div
                class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 cxrjfz"
                data-testid="heading-title"
              >
                

                <h6
                  class="sc-1k44dfv-0 eeeuqx"
                >
                  Sample Heading
                </h6>
              </div>
              <ul
                class="sc-1k44dfv-0 sc-1k44dfv-0 egEiUc"
                data-testid="product-features"
              >
                <li
                  class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                  color="typographyDefault"
                  data-testid="product-feature"
                  font-family="default"
                  font-weight="regular"
                >
                  Commercial lawn and garden equipment
                </li>
                <li
                  class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                  color="typographyDefault"
                  data-testid="product-feature"
                  font-family="default"
                  font-weight="regular"
                >
                  Construction/industrial equipment
                </li>
                <li
                  class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                  color="typographyDefault"
                  data-testid="product-feature"
                  font-family="default"
                  font-weight="regular"
                >
                  Agricultural equipment
                </li>
                <li
                  class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                  color="typographyDefault"
                  data-testid="product-feature"
                  font-family="default"
                  font-weight="regular"
                >
                  Water pumps
                </li>
              </ul>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 cDyBNH"
              >
                <a
                  class="Button-pulw6t-0 sc-1k44dfv-0 jVKOFV"
                  data-gtm-body-style="HorizontalCrankshaft"
                  data-gtm-component-type="EngineDetailsCard"
                  data-gtm-interaction-type="find distributor"
                  data-gtm-model="GX Series"
                  data-gtm-title="Find a distributor"
                  data-gtm-trim="GX100"
                  data-testid="primary-button"
                  href="Shared.Common.dealersPageUrl"
                >
                  Sample Text
                </a>
                <button
                  class="Button-pulw6t-0 sc-1k44dfv-0 WeFee"
                  data-gtm-body-style="HorizontalCrankshaft"
                  data-gtm-component-type="EngineDetailsCard"
                  data-gtm-interaction-type="comparison results"
                  data-gtm-model="GX Series"
                  data-gtm-title="item added"
                  data-gtm-trim="GX100"
                  data-testid="secondary-button"
                >
                  Shared.Common.addToCompareButton
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fresnel-container fresnel-lessThan-desktop "
      >
        <div
          class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 hFojy"
        >
          <div
            class="Box-sc-1bsf784-0 jGVkdm"
            display="flex"
            width="100%"
          >
            <div
              class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 sgibs"
              width="1,1,0.5"
            >
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 hQILzn"
              >
                <div
                  class="Box-sc-1bsf784-0 isOmPB sc-1k44dfv-0 ednPAD"
                  display="flex"
                >
                  <div
                    class="Box-sc-1bsf784-0 gaxtDi"
                    display="flex"
                    width="100%"
                  >
                    <div
                      class="Box-sc-1bsf784-0 Column-t2p57o-0 hVRYAI"
                      display="flex"
                      width="0,0.125"
                    />
                    <div
                      class="Box-sc-1bsf784-0 Column-t2p57o-0 jcWdVb"
                      display="flex"
                      width="1,0.75"
                    >
                      <div
                        class="Box-sc-1bsf784-0 sc-1k44dfv-0 jazOpQ"
                      >
                        <div
                          aria-live="polite"
                          class="Box-sc-1bsf784-0 sc-1k44dfv-0 daGqcZ"
                          height="157px,362px,402px"
                        >
                          <div
                            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jyKEsn"
                          >
                            <div
                              class="Box-sc-1bsf784-0 sc-1k44dfv-0 dlUJcO"
                              data-testid="product-image-container"
                            >
                              <img
                                alt="Front of Honda GX100 engine"
                                class="sc-1k44dfv-0 cFTYBj"
                                data-testid="product-image"
                                draggable="true"
                                fetchpriority="high"
                                loading="eager"
                                src="/sample-mock-data/media/img/product-card/engineImage.png"
                                width="auto"
                              />
                            </div>
                          </div>
                          <div
                            class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
                          >
                            <div
                              class="Box-sc-1bsf784-0 sc-1k44dfv-0 dlUJcO"
                              data-testid="product-image-container"
                            >
                              <img
                                alt="Side of Honda GX100 engine"
                                class="sc-1k44dfv-0 cFTYBj"
                                data-testid="product-image"
                                draggable="true"
                                fetchpriority="high"
                                loading="eager"
                                src="/sample-mock-data/media/img/product-details-card/engineDetailsImage.png"
                                width="auto"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="Box-sc-1bsf784-0 Column-t2p57o-0 hVRYAI"
                      display="flex"
                      width="0,0.125"
                    />
                  </div>
                </div>
                <div
                  class="Box-sc-1bsf784-0 sc-1k44dfv-0 elVxZQ"
                >
                  <div
                    aria-label="Shared.Common.previousSlideAria"
                    class="IconWrapper-lp9q2r-0 hIeNMk"
                    cursor="pointer"
                    data-testid="arrow-left"
                    display="flex"
                    role="button"
                    tabindex="0"
                  />
                  <div
                    class="Box-sc-1bsf784-0 sc-1k44dfv-0 evNJbO"
                  >
                    <button
                      aria-label="0"
                      class="sc-1k44dfv-0 cjSExO"
                      data-testid="step1"
                      role="button"
                    />
                    <button
                      aria-label="1"
                      class="sc-1k44dfv-0 ipUiMI"
                      data-testid="step2"
                      role="button"
                    />
                  </div>
                  <div
                    aria-label="Shared.Common.nextSlideAria"
                    class="IconWrapper-lp9q2r-0 hIeNMk"
                    cursor="pointer"
                    data-testid="arrow-right"
                    display="flex"
                    role="button"
                    tabindex="0"
                  />
                </div>
              </div>
            </div>
            <div
              class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 cVEbMC"
              width="1,1,0.5"
            >
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 krtBLR"
              >
                <h1
                  class="sc-1k44dfv-0 sc-1k44dfv-0 eNZGfe"
                  data-testid="product-title"
                >
                  GX100
                </h1>
                <div
                  class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 jVHFQZ"
                  data-testid="product-subtitle"
                >
                  

                  <h5
                    class="sc-1k44dfv-0 gXWcpP"
                  >
                    Horizontal Crankshaft
                  </h5>
                </div>
                <p
                  class="sc-1k44dfv-0 sc-1k44dfv-0 iZbHgV"
                  color="typographyDefault"
                  data-testid="tagline"
                  font-family="default"
                  font-weight="regular"
                >
                  <span>
                    Day in, day out, get solid performance, reliability and fuel efficiency. Ideal for lawn, construction and agriculture equipment.
                  </span>
                </p>
                <div
                  class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 cxrjfz"
                  data-testid="heading-title"
                >
                  

                  <h6
                    class="sc-1k44dfv-0 eeeuqx"
                  >
                    Sample Heading
                  </h6>
                </div>
                <ul
                  class="sc-1k44dfv-0 sc-1k44dfv-0 egEiUc"
                  data-testid="product-features"
                >
                  <li
                    class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                    color="typographyDefault"
                    data-testid="product-feature"
                    font-family="default"
                    font-weight="regular"
                  >
                    Commercial lawn and garden equipment
                  </li>
                  <li
                    class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                    color="typographyDefault"
                    data-testid="product-feature"
                    font-family="default"
                    font-weight="regular"
                  >
                    Construction/industrial equipment
                  </li>
                  <li
                    class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                    color="typographyDefault"
                    data-testid="product-feature"
                    font-family="default"
                    font-weight="regular"
                  >
                    Agricultural equipment
                  </li>
                  <li
                    class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 sc-1k44dfv-0 gaViIl"
                    color="typographyDefault"
                    data-testid="product-feature"
                    font-family="default"
                    font-weight="regular"
                  >
                    Water pumps
                  </li>
                </ul>
                <div
                  class="Box-sc-1bsf784-0 sc-1k44dfv-0 cDyBNH"
                >
                  <a
                    class="Button-pulw6t-0 sc-1k44dfv-0 jVKOFV"
                    data-gtm-body-style="HorizontalCrankshaft"
                    data-gtm-component-type="EngineDetailsCard"
                    data-gtm-interaction-type="find distributor"
                    data-gtm-model="GX Series"
                    data-gtm-title="Find a distributor"
                    data-gtm-trim="GX100"
                    data-testid="primary-button"
                    href="Shared.Common.dealersPageUrl"
                  >
                    Sample Text
                  </a>
                  <button
                    class="Button-pulw6t-0 sc-1k44dfv-0 WeFee"
                    data-gtm-body-style="HorizontalCrankshaft"
                    data-gtm-component-type="EngineDetailsCard"
                    data-gtm-interaction-type="comparison results"
                    data-gtm-model="GX Series"
                    data-gtm-title="item added"
                    data-gtm-trim="GX100"
                    data-testid="secondary-button"
                  >
                    Shared.Common.addToCompareButton
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
